### PR TITLE
Enforce max output sizes for both client + node

### PR DESCRIPTION
**Pull request by zpalmtree**
Offers separate limits for network and client, client limits ensures users do not create outputs that are unspendable, and network limit ensures outputs that are impossible to spend are not created.

### CODE REVIEW

File: src/config/CryptoNoteConfig.h [hunk 1]
```diff
@@ -162,6 +162,18 @@ namespace CryptoNote
 
         const uint64_t MAX_EXTRA_SIZE_V2_HEIGHT = 1300000;
 
+        /* 30 trillion atomic, or 300 billion TRTL */
+        /* This is enforced on the daemon side. An output > 300 billion causes
+         * an invalid block. */
+        const uint64_t MAX_OUTPUT_SIZE_NODE   = 300'000'000'000'00;
```
Comment (brandonlehmann) [inline src/config/CryptoNoteConfig.h]: Actually, > 250'000'000'000'00 will cause the same problem won't it?

Comment (zpalmtree) [reply src/config/CryptoNoteConfig.h]: `250'000'000'000'00` is an output of `200'000'000'000'00` and an output of `50'000'000'000'00` ;)

Comment (brandonlehmann) [reply src/config/CryptoNoteConfig.h]: how do you explain those outputs that aren't pretty amounts in the chain then mister? :)

Comment (zpalmtree) [reply src/config/CryptoNoteConfig.h]: touché

**Comment by ExtraHash** [comment]
What happens to the one already created 1 billion output, would this render it unspendable?

**Comment by zpalmtree** [comment]
> What happens to the one already created 1 billion output, would this render it unspendable?
Can you do math?

**Comment by brandonlehmann** [comment]
> What happens to the one already created 1 billion output, would this render it unspendable?

1bil < 100bil

**Comment by zpalmtree** [comment]
Removing WIP, have tested both client side and node side and it functions as I expected.

File: src/config/CryptoNoteConfig.h (hunk 1 above)
Comment (brandonlehmann) [reply src/config/CryptoNoteConfig.h]: I'll ask again -- what prevents someone from making a `non-pretty` output that exceeds 250bil?

Comment (zpalmtree) [reply src/config/CryptoNoteConfig.h]: Fixed

**Pull request merged**